wd_helper.au3

```diff
@@ -477,7 +477,7 @@ Func _WD_DebugSwitch($vMode = Default, $iErr = @error, $iExt = @extended)
 		$iResult = 1
 	ElseIf IsInt($vMode) And $vMode >= $_WD_DEBUG_None And $vMode <= $_WD_DEBUG_Full Then ; setting new debug level
 		ReDim $a_WD_DEBUG_SavedStack[$iStackSize + 1] ; resize / add new position to the stack
-		$a_WD_DEBUG_SavedStack[$iStackSize - 1] = $vMode ; set new last stack value
+		$a_WD_DEBUG_SavedStack[$iStackSize] = $vMode ; set new last stack value
```

Comment:
oh. I see my mistake.
After **ReDim** last element ID should not be decrased.
But will check it deeply in a moment.

Reply:
Please see my last commit, which fixes another issue with the above. Here's a modified version of your code that I used for testing --

```autoit
#include "..\webdriver\wd_helper.au3"

ConsoleWrite(' START: ' & $_WD_DEBUG & @CRLF)

_WD_DebugSwitch($_WD_DEBUG_None)
ConsoleWrite(' After $_WD_DEBUG_None: ' & $_WD_DEBUG & @CRLF)
_WD_DebugSwitch()
ConsoleWrite(' Restoring: ' & $_WD_DEBUG & @CRLF)

_WD_DebugSwitch($_WD_DEBUG_Error)
ConsoleWrite(' After $_WD_DEBUG_Error: ' & $_WD_DEBUG & @CRLF)

_WD_DebugSwitch($_WD_DEBUG_Info)
ConsoleWrite(' After $_WD_DEBUG_Info: ' & $_WD_DEBUG & @CRLF)
_WD_DebugSwitch()
ConsoleWrite(' Restoring: ' & $_WD_DEBUG & @CRLF)

_WD_DebugSwitch()
ConsoleWrite(' Restoring: ' & $_WD_DEBUG & @CRLF)

_WD_DebugSwitch($_WD_DEBUG_Full)
ConsoleWrite(' After $_WD_DEBUG_Full: ' & $_WD_DEBUG & @CRLF)
_WD_DebugSwitch()
ConsoleWrite(' Restoring: ' & $_WD_DEBUG & @CRLF)
```

and my results --

```
     START: 2
     After $_WD_DEBUG_None: 0
    _WD_DebugSwitch ==> Success [0 / 0]
     Restoring: 2
     After $_WD_DEBUG_Error: 1
    _WD_DebugSwitch ==> Success [0 / 0]
     After $_WD_DEBUG_Info: 2
     Restoring: 1
    _WD_DebugSwitch ==> Success [0 / 0]
     Restoring: 2
    _WD_DebugSwitch ==> Success [0 / 0]
     After $_WD_DEBUG_Full: 3
    _WD_DebugSwitch ==> Success [0 / 0]
     Restoring: 2
```